crates/mipsy_interactive/src/interactive/helper.rs

```diff
@@ -33,15 +53,47 @@ impl Completer for MyHelper {
         pos: usize,
         ctx: &Context<'_>,
     ) -> Result<(usize, Vec<Pair>), ReadlineError> {
-        self.completer.complete(line, pos, ctx)
+        self.completer.complete(line, pos, ctx).and_then(|x| {
+            // will be 0 for first arg. kinda a hacky way to check if we should match
+            // against files or not but thats okay.
+            // TODO: something like `match self.arg_type() { label => , file => number => }`
+            Ok(if x.0 == 0 {
+                (
+                    pos,
+                    self.closest(
+                        &[
+                            ctx.history()
+                                .iter()
+                                .rev()
+                                .map(|h| h.to_owned())
+                                .collect::<Vec<String>>(),
+                            self.defaults.clone(),
+                        ]
+                        .concat(),
+                        line,
+                        pos,
+                    )
+                    .iter()
+                    .map(|m| Pair {
+                        display: m.to_owned(),
```

Comment:
this should be the entire matched word instead of `match[pos..]` but since i dont even know if this is used and it doesnt seem that important i think its nicer than something like
```diff
-fn closest(&self, with: &Vec<String>, line: &str, pos: usize) -> Option<String>
+fn closest(&self, with: &Vec<String>, line: &str, pos: usize) -> Option<(&str, &str)>
````